src/test/java/org/sagebionetworks/bridge/spring/controllers/ParticipantDataControllerTest.java

```diff
@@ -0,0 +1,324 @@
+
+        assertEquals(result.getUserId(), participantData.getUserId());
+        assertEquals(result.getIdentifier(), participantData.getIdentifier());
+        assertEquals(result.getData(), participantData.getData());
+        assertEquals(result.getVersion(), participantData.getVersion());
+        assertSame(result, participantData);
+    }
+
+    @Test
+    public void testSaveDataForSelf() throws Exception {
+        String json = createJson("{'userId':'aUserId', 'data':{'field1':'a','field2':'b'}}");
+        mockRequestBody(mockRequest, json);
+
+        StatusMessage result = controller.saveDataForSelf(IDENTIFIER);
+        assertEquals(result.getMessage(), "Participant data saved.");
+
+        verify(mockParticipantDataService).saveParticipantData(anyString(), anyString(),
+                participantDataCaptor.capture());
+
+        ParticipantData capture = participantDataCaptor.getValue();
+
+        assertNull(capture.getUserId());
+        assertNull(capture.getIdentifier());
+        assertNull(capture.getVersion());
+        assertEquals(capture.getData(), participantData.getData());
+    }
+
+    @Test
+    public void testDeleteDataByIdentifier() {
+        StatusMessage result = controller.deleteDataByIdentifier(IDENTIFIER);
+        assertEquals(result.getMessage(), "Participant data record deleted.");
+
+        verify(mockParticipantDataService).deleteParticipantData(session.getId(), IDENTIFIER);
+    }
+
+    @Test
+    public void testGetAllDataForAdminWorker() {
+        when(mockAccountService.getAccount(OTHER_ACCOUNT_ID)).thenReturn(mockAccount);
+
+        ForwardCursorPagedResourceList<ParticipantData> expected = makeResults(OFFSET_KEY, PAGE_SIZE_INT);
+        doReturn(expected).when(mockParticipantDataService).getAllParticipantData(session.getId(), OFFSET_KEY, PAGE_SIZE_INT);
+
+        ForwardCursorPagedResourceList<String> result = controller.getAllDataForAdminWorker(session.getAppId(),
+                session.getId(), OFFSET_KEY, PAGE_SIZE_STRING);
+
+        assertEquals(expected.getItems().get(0).getIdentifier(), result.getItems().get(0));
+        assertEquals(expected.getItems().size(), result.getItems().size());
+
+        verify(mockParticipantDataService).getAllParticipantData(TEST_USER_ID, OFFSET_KEY, PAGE_SIZE_INT);
+    }
+
+    @Test
+    public void testDeleteAllParticipantDataForAdmin() {
+        doReturn(session).when(controller).getAuthenticatedSession(ADMIN);
+
+        StatusMessage result = controller.deleteAllParticipantDataForAdmin(session.getAppId(), session.getId());
+        assertEquals(result.getMessage(), "Participant data deleted.");
+
+        verify(mockParticipantDataService).deleteAllParticipantData(TEST_USER_ID);
+    }
+
+    @Test (expectedExceptions = UnauthorizedException.class)
+    public void testDeleteAllParticipantDataForAdminUnAuthorizedException() {
+        controller.deleteAllParticipantDataForAdmin(TEST_APP_ID, TEST_USER_ID);
+    }
```

Comment:
I don't think this test case actually tests our checkAdminSessionAppId() logic. Also, we need test cases for all the other admin APIs as well.